packages/js/ai/src/hooks/useBackgroundRemoval.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+
+export type BackgroundRemovalParams = {
+	imageFile: File;
+	returnPngImage?: boolean;
+	returnedImageSize?: string;
+	token: string;
+};
+
+type BackgroundRemovalResponse = {
+	loading: boolean;
+	error: Error | null;
+	imageData: Blob | null;
+	fetchImage: ( params: BackgroundRemovalParams ) => void;
+};
+
+const useBackgroundRemoval = (): BackgroundRemovalResponse => {
+	const [ loading, setLoading ] = useState( false );
+	const [ error, setError ] = useState< Error | null >( null );
+	const [ imageData, setImageData ] = useState< Blob | null >( null );
+
+	const fetchImage = async ( params: BackgroundRemovalParams ) => {
+		const { returnPngImage, returnedImageSize, imageFile, token } = params;
+
+		if ( ! token ) {
+			setError( new Error( 'Invalid token' ) );
```

Comment:
We have the `createExtendedError` utility in this package that allows you to inject more useful information into the error (such as `cause` and `code`). I found it useful in identifying errors when consuming the `useCompletion` hook.